salt/hosts/init.sls

```diff
@@ -1,28 +1,13 @@
-{%- set minealias    = salt['pillar.get']('hostsfile:alias', 'network.ip_addrs') %}
-{%- set minions      = salt['pillar.get']('hostsfile:minions', '*') %}
-{%- set pillar_hosts = salt['pillar.get']('hostsfile:hosts', {}) %}
-{%- set mine_hosts   = salt['mine.get'](minions, minealias) %}
-
-{%- set hosts = {} %}
-{%- if mine_hosts is defined %}
-{%-   do hosts.update(mine_hosts) %}
-{%- endif %}
-{%- do hosts.update(pillar_hosts) %}
-
-{%- for fqdn, addrlist in hosts.items() %}
-{{ fqdn }}-host-entry:
+# Internally we refer to hosts by their machine ID, which is also used as the minion ID.
+{%- for minion_id, addrlist in salt['mine.get']('*', 'network.ip_addrs').items() %}
+{{ minion_id }}-host-entry:
   host.present:
 {% if addrlist is string %}
     - ip: {{ addrlist }}
 {% else %}
     - ip: {{ addrlist|first }}
 {% endif %}
```

Comment:
I think this is what it currently does: it reads as _if `addrlist` is just a string (not a list), use `addrlist`, otherwise (it is a list), use the first IP in the list_